### PR TITLE
Cleanup System.Text.RegularExpressions tests

### PR DESCRIPTION
**Pull request by hughbe**
- Split test data and test code
- Use xunit's inlinedata and memberdata features
- Consolidate random test files into files with a better split

Follows up from #6515 
/cc @stephentoub

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @Priya91

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Release Build and Test please

File: src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+
+using System.Collections.Generic;
+using System.Globalization;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexGroupTests
+    {
+        private static readonly CultureInfo s_germanPhoneBookCulture = new CultureInfo("de-DE");
```
Comment (stephentoub) [inline src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: Is this field name correct?  Why isn't it just s_germanCulture?
cc: @tarekgh

**Comment by stephentoub** [comment]
@hughbe, generally the changes LGTM, but there's a small drop in the code coverage numbers.  Do you know why?  What got lost?

cc: @Priya91

**Comment by hughbe** [comment]
I'll take a look tonight

**Comment by Priya91** [comment]
I have been trying to figure out if all the deleted tests have be added in someother scenario elsewhere. Will pull this down locally to review.

**Comment by hughbe** [comment]
I've identified the cause of the decrease in coverage - I forgot to include an overload for `Regex.IsMatch(string input, string pattern)`
I ran the coverage runner locally: no change in line coverage, increase of 0.1% branch

**Comment by hughbe** [comment]
Rebased to fix a merge conflict

File: src/System.Text.RegularExpressions/tests/Regex.UnicodeChar.Tests.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                Assert.Equal(builder2.ToString(), match.Value);
+                Assert.Equal(0, match.Index);
+                Assert.Equal(validCharLength, match.Length);
+
+                match = match.NextMatch();
+                do
+                {
+                    // This is tedious. But we report empty Matches for each of the non-matching characters!!!
+                    // duh!!! because we say so on the pattern - remember what * stands for :-)
+                    Assert.Equal(string.Empty, match.Value);
```
Comment (Priya91) [inline src/System.Text.RegularExpressions/tests/Regex.UnicodeChar.Tests.cs]: Can you cleanup the comment to be,

```We get empty matches for each of the non-matching characters of input to match the * wildcard in regex pattern.```` 

or something more in third person. Thanks!

**Comment by hughbe** [comment]
Sorry - I had to rebase again to fix another merge conflict ;P

**Comment by hughbe** [comment]
Test Innerloop OSX Debug Build and Test
Test Innerloop OSX Release Build and Test

**Comment by Priya91** [comment]
LGTM

**Pull request merged**